### ISSUE TITLE
What is the LambdaWeights for lambdarank?

### ISSUE THREAD

**Issue by MoyanZitto** [tensorflow/ranking#39]
Hi authors, 
    Please correct me if I am wrong. In my opinion:
    (1) the lambdaloss paper "The LambdaLoss Framework for ranking metric optimization" offers an unified framework for metric-driven optimization: metric-driven loss can be expressed as sum of weighted loss for data pairs.
    (2) The weights in (1) is LambdaWeights class in tf-ranking code.
    (3) So, to make a metric-driven loss, all I need to do are:
           1. Subclass LambdaWeights and implement my custom weights
           2, Create a loss with my LambdaWeights object.
 
e.g: let's say I want to make a ranknet loss:
     (1) Instancelize a DCGLambdaWeight obeject.
     (2) Call _pariwise_logistic_loss, where the parameter lambda_weight is set to object in (1).

Is that correct?

**Comment by xuanhuiwang** [comment]
The flow looks right.

The LambdaRank weight is already implemented in tf-ranking. You can set the smooth_fraction=1.0 to use it.

To create a loss, a better way is to use the factory method: https://github.com/tensorflow/ranking/blob/master/tensorflow_ranking/python/losses.py#L53.

In that sense, you don't need to create a new DCGLambdaWeight, unless you want to try something else.

Also, when lambda_weight is enabled, you may need to re-tune your learning rate.

**Comment by MoyanZitto** [comment]
@xuanhuiwang 
Thanks for your kindly reply, it is really helpful.
